di-ipv-evcs-stub/lambdas/src/services/evcsService.ts

```diff
@@ -0,0 +1,195 @@
+import {
+  DynamoDB,
+  QueryInput,
+  PutItemInput,
+  UpdateItemInput,
+  PutItemCommandOutput,
+  UpdateItemCommandOutput,
+} from "@aws-sdk/client-dynamodb";
+import { marshall, unmarshall } from "@aws-sdk/util-dynamodb";
+
+import PostRequest from "../domain/postRequest";
+import ServiceResponse from "../domain/serviceResponse";
+import { VcState } from "../domain/enums/vcState";
+import EvcsVcItem from "../model/evcsVcItem";
+
+import { config } from "../common/config";
+import { getSsmParameter } from "../common/ssmParameter";
+import { v4 as uuid } from "uuid";
+import PatchRequest from "../domain/patchRequest";
+import VCProvenance from "../domain/enums/vcProvenance";
+
+const dynamoClient = config.isLocalDev
+  ? new DynamoDB({
+      endpoint: config.localDynamoDbEndpoint,
+      region: config.region,
+    })
+  : new DynamoDB({ region: config.region });
+
+export async function processPostUserVCsRequest(
+  userId: string,
+  postRequest: PostRequest[],
+): Promise<ServiceResponse> {
+  let response: ServiceResponse = {
+    response: {},
+  };
+  try {
+    console.info(`Post user record.`);
+    const allPromises: Promise<PutItemCommandOutput>[] = [];
+    const ttl = await getTtl();
+    for (const postRequestItem of postRequest) {
+      const vcItem: EvcsVcItem = {
+        userId,
+        vc: postRequestItem.vc,
+        vcSignature: postRequestItem.vc.split(".")[2],
+        state: postRequestItem.state,
+        metadata: postRequestItem.metadata!,
```

Comment:
In the `PostRequest` interface `metadata` is marked as optional, but here we're asserting it's not null. If it ever is I think we'd get a runtime error. Should it have a default of empty object instead?

Reply:
done